src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,200 @@
+            };
+
+            Put["/redirects/{messageredirectid:guid}/"] = parameters =>
+            {
+                Guid messageRedirectId = parameters.messageredirectid;
+
+                var request = this.Bind<MessageRedirectRequest>();
+
+                if (string.IsNullOrWhiteSpace(request.tophysicaladdress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirects = MessageRedirectsCollection.GetOrCreate(session);
+
+                    var messageRedirect = redirects[messageRedirectId];
+
+                    if (messageRedirect == null)
+                    {
+                        return HttpStatusCode.NotFound;
+                    }
+
+                    var toMessageRedirectId = DeterministicGuid.MakeId(request.tophysicaladdress.ToLowerInvariant());
+
+                    if (redirects[toMessageRedirectId] != null)
+                    {
+                        return HttpStatusCode.Conflict;
+                    }
+
+                    var messageRedirectChanged = new MessageRedirectChanged
+                    {
+                        MessageRedirectId = messageRedirectId,
+                        PreviousToPhysicalAddress = messageRedirect.ToPhysicalAddress,
+                        FromPhysicalAddress = messageRedirect.FromPhysicalAddress,
+                        ToPhysicalAddress = messageRedirect.ToPhysicalAddress = request.tophysicaladdress.ToLowerInvariant(),
+                    };
+
+                    messageRedirect.LastModifiedTicks = DateTime.UtcNow.Ticks;
+
+                    redirects.Save(session);
+
+                    Bus.Publish(messageRedirectChanged);
+
+                    return HttpStatusCode.OK;
+                }
+            };
+
+            Delete["/redirects/{messageredirectid:guid}/"] = parameters =>
+            {
+                Guid messageRedirectId = parameters.messageredirectid;
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirects = MessageRedirectsCollection.GetOrCreate(session);
+
+                    var messageRedirect = redirects[messageRedirectId];
+
+                    if (messageRedirect == null)
+                    {
+                        return HttpStatusCode.OK;
+                    }
+
+                    redirects.Redirects.Remove(messageRedirect);
+
+                    redirects.Save(session);
+
+                    Bus.Publish<MessageRedirectRemoved>(evt =>
+                    {
+                        evt.MessageRedirectId = messageRedirectId;
+                        evt.FromPhysicalAddress = messageRedirect.FromPhysicalAddress;
+                        evt.ToPhysicalAddress = messageRedirect.ToPhysicalAddress;
+                    });
+                }
+
+                return HttpStatusCode.OK;
```

Comment:
As you stated in slack both are acceptable, but I think 204 is the more appropriate.